supertokens/supertokens.go

```diff
@@ -26,19 +26,26 @@ func newSuperTokens(config TypeInput) (*SuperTokens, error) {
 		return nil, err
 	}
 
-	hostList := strings.Split(config.Supertoken.ConnectionURI, ";")
-	var hosts []NormalisedURLDomain
-	for _, h := range hostList {
-		host, err := NewNormalisedURLDomain(h, false)
-		if err != nil {
-			return nil, err
+	q := Querier{}
+	if config.Supertoken != nil {
+		hostList := strings.Split(config.Supertoken.ConnectionURI, ";")
+		var hosts []NormalisedURLDomain
+		for _, h := range hostList {
+			host, err := NewNormalisedURLDomain(h, false)
+			if err != nil {
+				return nil, err
+			}
+			hosts = append(hosts, *host)
+		}
+		if config.Supertoken.APIKey != nil {
+			q.Init(hosts, *config.Supertoken.APIKey)
+		} else {
+			q.Init(hosts, "")
 		}
-		hosts = append(hosts, *host)
+	} else {
+		q.Init(nil, "")
```

Comment:
- The init function should be a static function, so you should not have to do `q.Init(..)`
- Also, I prefer if you make `apiKey` in it a pointer so that it's more type safe to use it.